### PR TITLE
Do not retrieve port for containers not found in cache.

### LINKED ISSUE #7576: docker persona crashes with panic while trying to get a list of containers.
```
2018-03-22 19:24:09.437254 I | http: panic serving 10.84.232.126:50910: runtime error: invalid memory address or nil pointer dereference
goroutine 314 [running]:
net/http.(*conn).serve.func1(0xc420d235e0)
	/usr/local/go/src/net/http/server.go:1726 +0xd0
panic(0x1a8f220, 0x2ffcbf0)
	/usr/local/go/src/runtime/panic.go:505 +0x229
github.com/vmware/vic/lib/apiservers/engine/network.PortForwardingInformation(0x0, 0xc4201f6d60, 0x1, 0x1, 0x0, 0x1, 0x8)
	/go/src/github.com/vmware/vic/lib/apiservers/engine/network/utils.go:567 +0x667
github.com/vmware/vic/lib/apiservers/engine/backends.(*ContainerBackend).Containers(0xc4204ac930, 0xc4206a2a40, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/vmware/vic/lib/apiservers/engine/backends/container.go:1405 +0x1005
github.com/vmware/vic/vendor/github.com/docker/docker/api/server/router/container.(*containerRouter).getContainersJSON(0xc4206f0100, 0x7f7cfb74c5b0, 0xc420012960, 0x2025880, 0xc42031db20, 0xc420188400, 0xc4200128d0, 0x2027b00, 0xc420012960)
	/go/src/github.com/vmware/vic/vendor/github.com/docker/docker/api/server/router/container/container_routes.go:50 +0x201
github.com/vmware/vic/vendor/github.com/docker/docker/api/server/router/container.(*containerRouter).(github.com/vmware/vic/vendor/github.com/docker/docker/api/server/router/container.getContainersJSON)-fm(0x7f7cfb74c5b0, 0xc420012960, 0x2025880, 0xc42031db20, 0xc420188400, 0xc4200128d0, 0x7f7cfb74c5b0, 0xc420012960)
	/go/src/github.com/vmware/vic/vendor/github.com/docker/docker/api/server/router/container/container.go:44 +0x69
github.com/vmware/vic/vendor/github.com/docker/docker/api/server/middleware.VersionMiddleware.WrapHandler.func1(0x7f7cfb74c5b0, 0xc420012930, 0x2025880, 0xc42031db20, 0xc420188400, 0xc4200128d0, 0x402875, 0x1ff4510)
	/go/src/github.com/vmware/vic/vendor/github.com/docker/docker/api/server/middleware/version.go:47 +0x440
github.com/vmware/vic/vendor/github.com/docker/docker/api/server/middleware.DebugRequestMiddleware.func1(0x7f7cfb74c5b0, 0xc420012930, 0x2025880, 0xc42031db20, 0xc420188400, 0xc4200128d0, 0x7f7cfb74c5b0, 0xc420012930)
	/go/src/github.com/vmware/vic/vendor/github.com/docker/docker/api/server/middleware/debug.go:22 +0x161
github.com/vmware/vic/vendor/github.com/docker/docker/api/server.(*Server).makeHTTPHandler.func1(0x2025880, 0xc42031db20, 0xc420188400)
	/go/src/github.com/vmware/vic/vendor/github.com/docker/docker/api/server/server.go:139 +0x1c6
net/http.HandlerFunc.ServeHTTP(0xc4207812c0, 0x2025880, 0xc42031db20, 0xc420188400)
	/usr/local/go/src/net/http/server.go:1947 +0x44
github.com/vmware/vic/vendor/github.com/docker/docker/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc420d35720, 0x2025880, 0xc42031db20, 0xc420188400)
	/go/src/github.com/vmware/vic/vendor/github.com/docker/docker/vendor/github.com/gorilla/mux/mux.go:103 +0x226
github.com/vmware/vic/vendor/github.com/docker/docker/api/server.(*routerSwapper).ServeHTTP(0xc42047d960, 0x2025880, 0xc42031db20, 0xc420188400)
	/go/src/github.com/vmware/vic/vendor/github.com/docker/docker/api/server/router_swapper.go:29 +0x70
net/http.serverHandler.ServeHTTP(0xc420d52a90, 0x2025880, 0xc42031db20, 0xc420188400)
	/usr/local/go/src/net/http/server.go:2694 +0xbc
net/http.(*conn).serve(0xc420d235e0, 0x2027a40, 0xc4206a2900)
	/usr/local/go/src/net/http/server.go:1830 +0x651
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:2795 +0x27b
```

### PR DESCRIPTION
**Pull request by vburenin**
Fixes #7576

### CODE REVIEW

File: lib/apiservers/engine/network/utils.go [hunk 1]
```diff
@@ -564,7 +564,7 @@ func PortForwardingInformation(vc *viccontainer.VicContainer, ips []string) []ty
 	//c := cache.ContainerCache().GetContainer(cid)
```
Comment (gigawhitlocks) [inline lib/apiservers/engine/network/utils.go]: I know this isn't your code but can we delete these commented-out lines?

File: lib/apiservers/engine/network/utils.go [hunk 2]
```diff
@@ -564,7 +564,7 @@ func PortForwardingInformation(vc *viccontainer.VicContainer, ips []string) []ty
...
 
 	if vc == nil {
-		log.Errorf("Could not find container with ID %s", vc.ContainerID)
+		log.Errorf("No container provided")
```
Comment (gigawhitlocks) [inline lib/apiservers/engine/network/utils.go]: I presume the context that was in `vc.ContainerID` that we're no longer logging here is the same as the variable `t.ContainerConfig.ContainerID` at the top of the diff? If it is not, is there a reason we're eliminating this context from the logs?

Comment (vburenin) [reply lib/apiservers/engine/network/utils.go]: How can you log vc.ContainerID if vc is nil? :)

File: lib/apiservers/engine/network/utils.go (hunk 1 above)
Comment (vburenin) [reply lib/apiservers/engine/network/utils.go]: nope ;)

**Comment by sflxn** [comment]
lgtm.  This reminds me I have to back port some cases similar to this from the wolfpack branch.

File: lib/apiservers/engine/network/utils.go (hunk 2 above)
Comment (gigawhitlocks) [reply lib/apiservers/engine/network/utils.go]: LOL good question 😆

Comment (gigawhitlocks) [reply lib/apiservers/engine/network/utils.go]: NEVERMIND

**Pull request merged**